code-challenges.test.js

```diff
@@ -15,42 +15,195 @@
+//         // Expected output: "L4ck4d41s1c4l"
+//         const secretCodeWord2 = "Gobbledygook"
+//         // Expected output: "G0bbl3dyg00k"
+//         const secretCodeWord3 = "Eccentric"
+//         // Expected output: "3cc3ntr1c"
+//         expect(stringCode(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+//         expect(stringCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+//         expect(stringCode(secretCodeWord3)).toEqual("3cc3ntr1c")
 
+//     })
+// })
+
+//* RAN TEST (RESULTS BELOW): *//
+
+// ReferenceError: stringCode is not defined
 
 // b) Create the function that makes the test pass.
 
+//Goal: Takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+
+//Pseudocode:
+// Create a function expression that takes a string as a parameter. (stringCode)
+// Use for loop and .length property to setup function iterations.
+// Have the function invoke the replace method on the given parameter thru the created variable (secretString), to replace the given vowel indexes to integers. 
+// Have the function return the expected output. 
+
+// const stringCode= (secretString) => {
+//     for (let i=0; i <secretString.length; i++) {
+//         if (secretString[i] = "a") {
+//             return secretString[i].replace("a",4) 
+//         }
+//         else if(secretString[i] = "e") {
+//             return secretString[i].replace("e",3)
+//         }
+//         else if(secretString[i] = "i") {
+//             return secretString[i].replace("i",1)
+//         }
+//         else if(secretString[i] = "o") {
+//             return secretString[i].replace("o",0)
+//         } else {
+//             return secretString[i]
+//         }
+//     }
+// }
+
+//* RAN TEST (RESULTS BELOW): *//
+
+// FAIL  ./code-challenges.test.js
+// stringCode
+//   ✕ takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0 (2 ms)
+
+// ● stringCode › takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0
+
+//   expect(received).toEqual(expected) // deep equality
+
+//   Expected: "L4ck4d41s1c4l"
+//   Received: "L"
+
+//     24 |         const secretCodeWord3 = "Eccentric"
+//     25 |         // Expected output: "3cc3ntr1c"
+//   > 26 |         expect(stringCode(secretCodeWord1)).toEqual         ("L4ck4d41s1c4l")
+//        |                                             ^
+//     27 |         expect(stringCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+//     28 |         expect(stringCode(secretCodeWord3)).toEqual("3cc3ntr1c")
+//     29 |
+
+// ** NOTES TO INSTRUCTOR **: I'm not to sure why I don't have a passing test or what Jest expecting from me with this one, it seems to have an issue with my Line #26 (as shown in the error message section above). Found this very frustrating to say the least (this is why my code is commented out).
+
```

Comment:
Hi Steven - your code looks great!  Really good job with the pseudocode and logic.  The problem is you are only returning a single letter, not the whole string. Also, you need to account for capital letters.  Your thought process in the right direction.  I encourage you to try refactoring.  Let me know if you have questions.